test/uk/gov/hmrc/hecstubs/controllers/IndividualAccountOverviewControllerSpec.scala

```diff
@@ -0,0 +1,221 @@
+
+  def fakeRequest(environment: String, correlationId: String) = FakeRequest(
+    method = "GET",
+    uri = "/",
+    headers = FakeHeaders(
+      Seq(("Content-type", "application/json"), ("Environment", environment), ("CorrelationId", correlationId))
+    ),
+    body = AnyContentAsEmpty
+  )
+
+  private val controller = new IndividualAccountOverviewController(mockCC)
+
+  val validUtr   = "1234567890"
+  val inValidUtr = List("12345678901", "ABC1234567", "AB675^^&hg")
+
+  val validTaxYear   = "2021"
+  val inValidTaxYear = List("202", "20188", "12", "0", "")
+
+  "IndividualAccountOverviewController" when {
+
+    "fetching account overview details " should {
+
+      "return 200 and return the correct response" in {
+
+        val expectedJson: JsValue  = Json.parse("""
+            |{
+            |    "utr": "1234567890",
+            |    "taxYear": "2021",
+            |    "returnStatus": "Return Found"
+            |}
+            |""".stripMargin)
+        val result: Future[Result] =
+          controller.individualAccountOverview(validUtr, validTaxYear)(fakeRequest("live", UUID.randomUUID().toString))
+        status(result) shouldBe Status.OK
+        contentAsJson(result) mustBe expectedJson
+      }
+
+      "return bad request" when {
+
+        "One invalid parameter" when {
+
+          "utr is Invalid" in {
+
+            val expectedJson: JsValue = badJsonResponse("INVALID_UTR", "Invalid parameter utr.")
+
+            inValidUtr.map { invUtr =>
+              val result = controller.individualAccountOverview(invUtr, validTaxYear)(
+                fakeRequest("live", UUID.randomUUID().toString)
+              )
+              status(result) shouldBe Status.BAD_REQUEST
+              contentAsJson(result) mustBe expectedJson
+            }
+
+          }
+
+          "taxYear is Invalid" in {
+
+            val expectedJson: JsValue = badJsonResponse("INVALID_TAXYEAR", "Invalid parameter taxYear.")
+            inValidTaxYear.map { inTaxYear =>
+              val result = controller.individualAccountOverview(validUtr, inTaxYear)(
+                fakeRequest("live", UUID.randomUUID().toString)
+              )
+              status(result) shouldBe Status.BAD_REQUEST
+              contentAsJson(result) mustBe expectedJson
+            }
+
+          }
+
+          "Wrong Environment value is  passed in the header" in {
+            val result: Future[Result] =
+              controller.individualAccountOverview(validUtr, validTaxYear)(
+                fakeRequest("env", UUID.randomUUID().toString)
+              )
+            status(result) shouldBe Status.BAD_REQUEST
+
+          }
+
+          "No Environment value is  passed in the header" in {
+            val result: Future[Result] =
+              controller.individualAccountOverview(validUtr, validTaxYear)(
+                fakeRequest("", UUID.randomUUID().toString)
+              )
+            status(result) shouldBe Status.BAD_REQUEST
+
```

Comment:
do we need another test for when the environment value is missing instead of being an empty string?

Reply:
Yes, will add that once Ian confirm the error message to be displayed

Reply:
added new test case as of now